src/Umbraco.Forms.Integrations.Automation.Zapier/Services/ZapierFormSubscriptionHookService.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+using Umbraco.Forms.Integrations.Automation.Zapier.Models.Dtos;
+
+#if NETCOREAPP
+using Microsoft.Extensions.Logging;
+
+using Umbraco.Cms.Core.Scoping;
+#else
+using Umbraco.Core.Logging;
+using Umbraco.Core.Scoping;
+#endif
+
+namespace Umbraco.Forms.Integrations.Automation.Zapier.Services
+{
+    public class ZapierFormSubscriptionHookService
+    {
+        private readonly IScopeProvider _scopeProvider;
+
+#if NETCOREAPP
+        private readonly ILogger<ZapierFormSubscriptionHookService> _logger;
+
+        public ZapierFormSubscriptionHookService(IScopeProvider scopeProvider, ILogger<ZapierFormSubscriptionHookService> logger)
+        {
+            _scopeProvider = scopeProvider;
+
+            _logger = logger;
+        }
+#else
+        private readonly ILogger _logger;
+
+         public ZapierFormSubscriptionHookService(IScopeProvider scopeProvider, ILogger logger)
+        {
+            _scopeProvider = scopeProvider;
+
+            _logger = logger;
+        }
+#endif
+
+        public bool TryGetByName(string formName, out IEnumerable<FormConfigDto> dto)
+        {
+            using (var scope = _scopeProvider.CreateScope())
+            {
+                var entities =
+                    scope.Database
+                        .Query<FormConfigDto>( "SELECT * FROM zapierFormSubscriptionHook where FormName = @0", formName)
```

Comment:
Couple of things here.

1. We should be looking up and storing form Id here I think, rather than form name, as the editor can change the name at any time.
2. Do we actually need a separate table for forms?  I was thinking we just have one table for all subscriptions that has three fields:
 - Id
 - EntityId (which would be a string, and we'd store here the content type alias for content and the form's GUID Id for forms)
 - HookUrl

Reply:
Updated.